partiql-tests-data/eval/query/limitoffset.ion

```diff
@@ -0,0 +1,285 @@
+      output:$bag::[
+        {'a': 2}, {'a': 3}, {'a': 4}, {'a': 5}
+      ]
+    }
+  },
+
+  // LIMIT 10 and OFFSET 10 should return no result
+  {
+    name:"limit 10 offset 10",
+    statement:"SELECT * FROM foo LIMIT 10 OFFSET 10",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[]
+    }
+  },
+
+  // LIMIT 2 and OFFSET 2 should return third and fourth results
+  {
+    name:"limit 2 offset 2",
+    statement:"SELECT * FROM foo GROUP BY a LIMIT 2 OFFSET 2",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[
+        {'a': 3}, {'a': 4}
+      ],
+    }
+  },
+
+  // LIMIT and OFFSET applied after GROUP BY
+  {
+    name:"limit offset after group by",
+    statement:"SELECT * FROM foo GROUP BY a LIMIT 1 OFFSET 1",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[
+        {'a': 2}
+      ],
+    }
+  },
+
+  // OFFSET value can be subtraction of 2 numbers
+  {
+    name:"offset 2-1",
+    statement:"SELECT * FROM foo OFFSET 2 - 1",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[
+        {'a': 2}, {'a': 3}, {'a': 4}, {'a': 5}
+      ]
+    }
+  },
+
+  // OFFSET value can be addition of 2 numbers
+  {
+    name:"offset 2+1",
+    statement:"SELECT * FROM foo OFFSET 2 + 1",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[
+        {'a': 4}, {'a': 5}
+      ]
+    }
+  },
+
+  // OFFSET value can be multiplication of 2 numbers
+  {
+    name:"offset 2*1",
+    statement:"SELECT * FROM foo OFFSET 2 * 1",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[
+        {'a': 3}, {'a': 4}, {'a': 5}
+      ]
+    }
+  },
+
+  // OFFSET value can be division of 2 numbers
+  {
+    name:"offset 2/1",
+    statement:"SELECT * FROM foo OFFSET 4 / 2",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[
+        {'a': 3}, {'a': 4}, {'a': 5}
+      ]
+    }
+  },
+
+  // OFFSET with GROUP BY and HAVING
+  {
+    name:"offset group by having",
+    statement:"SELECT * FROM foo GROUP BY a HAVING a > 2 LIMIT 1 OFFSET 1",
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:$bag::[
+        {'a': 4}
+      ],
+    }
+  },
+
+  // OFFSET with PIVOT
+  {
+    name:"offset with pivot",
+    statement:'''
+      PIVOT foo.a AT foo.b
+      FROM <<{'a': 1, 'b':'I'}, {'a': 2, 'b':'II'}, {'a': 3, 'b':'III'}>> AS foo
+      LIMIT 1 OFFSET 1
+    ''',
+    assert:{
+      evalMode:[EvalModeCoerce, EvalModeError],
+      result:EvaluationSuccess,
+      output:{'II': 2},
+    }
+  },
+
+  // OFFSET -1 should throw exception
```

Comment:
In this in `EvalErrormode` only i.e. in `EvalModeCoerce` should succeed?

Reply:
I added an issue for this: https://github.com/partiql/partiql-tests/issues/65